server/src/main/java/org/opensearch/indices/IndicesService.java

```diff
@@ -769,6 +764,9 @@ private EngineFactory getEngineFactory(final IndexSettings idxSettings) {
             if (idxSettings.isSegRepEnabled()) {
                 return new NRTReplicationEngineFactory();
             }
+            if (IndexModule.Type.REMOTE_SNAPSHOT.match(idxSettings)) {
+                return config -> new ReadOnlyEngine(config, new SeqNoStats(0, 0, 0), new TranslogStats(), true, Function.identity(), false);
```

Comment:
Does this need its own factory? I'd imagine Primaries still require IE? 

Also out of scope for this change but calling it out here. I'd imagine we want clusters with a mix of remote search shards + regular with either replication strategy.  This will restrict those regular shards from using segrep.  Thinking we need to combine these into a single factory and/or set the engine type based on the storage strategy at a shard lvl.

Reply:
Ignore me, that is phase 4 on the roadmap.

Reply:
Yeah, things are pretty simple here with searchable snapshots because all shards in the index are read-only remote shards. This definitely will get more complicated in the later phases.